controller/scheduler/scheduler.go

```diff
@@ -1035,6 +1274,10 @@ func (s *Scheduler) HandleInternalStateRequest(req *InternalStateRequest) {
 		req.State.Formations[key.String()] = &f
 	}
 
+	for id, vol := range s.volumes {
+		req.State.Volumes[id] = &(*vol)
```

Comment:
We need to copy it to create a "snapshot" of the scheduler's state to pass back to the caller.